includes/User.php

```diff
@@ -2174,7 +2174,9 @@ public function setPassword( $str ) {
 		}
 
 		$this->setInternalPassword( $str );
-
+		if($clearTokens) {
+			wfRunHooks('SetPassword', [$this->getId()]);
```

Comment:
Ok.

To follow the naming convention used in this class let's call this hook `UserSetPassword` or even `UserClearTokens` (the first name is a bit confusing as it's not triggered on every password set).